src/transformers/models/cohere2/modular_cohere2.py

```diff
@@ -351,6 +348,7 @@ def __init__(self, config: Cohere2Config, layer_idx: int):
         self.is_sliding = (layer_idx + 1) % self.config.sliding_window_pattern != 0
         self.sliding_window = config.sliding_window
 
+    @deprecate_kwarg("last_cache_position", version="4.53.0")
```

Comment:
Modular files inherit decorators for overwritten methods unless you add any new decorator on top of it; that's why you have decorators disappeared in modeling file

Reply:
The solution is to manually add all disappeared decorators 🥲